tests/lib/rules/no-shadow.js

```diff
@@ -91,6 +93,16 @@ eslintTester.addRuleTest("lib/rules/no-shadow", {
             errors: [{ message: "x is already declared in the upper scope.", type: "Identifier"}]
         },
         {
+            code: "{ let x = 2; } var x = 1;",
+            ecmaFeatures: {blockBindings: true},
+            errors: [{ message: "x is already declared in the upper scope.", type: "Identifier"}]
+        },
+        {
+            code: "function a() { let x = 2; } let x = 1;",
```

Comment:
Hm, my first thought is:

```js
function a() { consle.log(x); }
let x = "hello";
a(); // print "hello"
```

so declarations in a function in the TDZ can shadow outer declarations.
But now, I think it's bad usage (`no-use-before-define`?), so I will make this into valid.